### ISSUE TITLE
Text spacing should not be respected within raw environment

### ISSUE THREAD

**Issue by nmayorov** [typst/typst#3848]
### Description

Not sure if it can be classified as a bug, but I believe the behavior is undesirable.

Raw blocks are sometimes used to draw simple ASCII schemes. For example I've written a NMEA 0183 message format like
```
       1   2
       |   |
$GNHDT,x.x,T*hh<CR><LF>
```
Where "1" and "2" mark message fields. If I then use `set text(spacing: 150%)`, the item will break as "1" and "2" (considered as separate words, I believe) will be further separated. 

In a raw block each symbol (including spaces) must be displayed with the same size. It is even said in the documentation: "Displays the text verbatim and in a monospace font."

### Operating system

Linux

### Typst version

- [X] I am using the latest version of Typst

**Comment by Leedehai** [comment]
Agreed. Like #3454, users rightfully expect the spacing within the raw blocks should not be affected by (almost all) set/show rules.

**Comment by Leedehai** [comment]
Reasonable. Like https://github.com/typst/typst/issues/3454, users rightfully expect the spacing within the raw blocks should not be affected by (almost all) set/show rules.
